testutil/integration/suites/update_params.go

```diff
@@ -0,0 +1,424 @@
+//go:build integration
+
+package suites
+
+import (
+	"encoding/hex"
+	"reflect"
+	"testing"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/app/volatile"
+	"github.com/pokt-network/poktroll/testutil/cases"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	gatewaytypes "github.com/pokt-network/poktroll/x/gateway/types"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+	tokenomicstypes "github.com/pokt-network/poktroll/x/tokenomics/types"
+)
+
+// ParamType is a type alias for a module parameter type. It is the string that
+// is returned when calling reflect.Value#Type()#Name() on a module parameter.
+type ParamType = string
+
+const (
+	ParamTypeInt64   ParamType = "int64"
+	ParamTypeUint64  ParamType = "uint64"
+	ParamTypeFloat32 ParamType = "float32"
+	ParamTypeString  ParamType = "string"
+	ParamTypeBytes   ParamType = "uint8"
+	ParamTypeCoin    ParamType = "Coin"
+)
+
+const (
+	MsgUpdateParamsName = "MsgUpdateParams"
+	MsgUpdateParamName  = "MsgUpdateParam"
+)
+
+// ModuleParamConfig holds a set of valid parameters and type information for a
+// given module. It uses any instead of cosmostypes.Msg to mitigate easy mistakes
```

Comment:
We do use concrete types when **assigning** to this structure but cannot use concrete types in its definition because:
1. each module defines its own update message types and
2. this structure is intended to be used for each module (will hold each module's distinct concrete types). 

Making this structure generic is also not really feasible because it would require a type parameter for each msg field.

Upon further reflection, I don't think making this point in the comment is necessary. I've refactored the structure and updated the comments.